src/province/mod.rs

```diff
@@ -228,6 +228,45 @@ pub enum IranProvince {
     Zanjan,
 }
 
+impl From<&str> for IranProvince {
```

Comment:
I'm almost sure that there is a derive macro for it! 🤔 cannot remember the name.

Reply:
ooh yes strum crate good idea